src/robotFactory.js

```diff
@@ -1,10 +1,98 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    position = {
+      x: 0,
+      y: 0,
+    },
+    chipVersion,
+  ) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: position.x || 0,
+      y: position.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(
+    name,
+    weight,
+    position = {
+      x: 0,
+      y: 0,
+      z: 0,
+    },
+    chipVersion,
+  ) {
+    super(name, weight, position, chipVersion);
+    this.coords.z = position.z || 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    position = {
+      x: 0, y: 0, z: 0,
+    },
+    chipVersion,
+    maxLoadWeight,
+    currentLoad = null,
+  ) {
```

Comment:
Same as in prev robots
```suggestion
  constructor(
    name,
    weight,
    position,
    chipVersion,
    maxLoadWeight,
    currentLoad = null,
  ) {
```